File: programmers/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int time=0;
    int job=0;
    int pFlag=0;
    while(pFlag<progresses.size()){
        if(progresses[pFlag]+(speeds[pFlag]*time)>=100){
            for(int i=pFlag; i<progresses.size();i++){
                if(progresses[pFlag]+(speeds[pFlag]*time)>=100){
                    job++;
                    pFlag++;
                }
                else{
                    answer.push_back(job);
                    job=0;
                    break;
                }
            }
        }
        time++;
    }
    answer.push_back(job);
    return answer;
}

File: programmers/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int flag = 0;
        ArrayList<Integer> ans = new ArrayList<Integer>();
        while (flag != speeds.length) {
            for (int i = flag; i < progresses.length; i++) {
                progresses[i] += speeds[i];
            }
            if (progresses[flag] >= 100) {
                int cnt = 0;
                for (int i = flag; i < progresses.length; i++) {
                    if (progresses[i] >= 100) {
                        cnt++;
                        flag = i + 1;
                    } else
                        break;
                }
                ans.add(cnt);
            }
        }
        int[] answer = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++)
            answer[i] = (int) ans.get(i);


        return answer;
    }
}